File: insertion_sort/insertion_sort.cs
public static void InsertionSort(int[] arr)
{
    for (int i = 1; i < arr.Length; i++)
    {
        int key = arr[i]; // The element to be inserted in the sorted part of the array
        int j = i - 1; // Index of the last element in the sorted part of the array

        while (j >=0 && arr[j]> key)
        {
            arr[j + 1] = arr[j]; // Shift the element to the right
            j--;
        }
        arr[j + 1] = key; // Insert the key in its correct position

        Console.WriteLine(string.Join(" ", arr));
    }
}

public static void Main(string[] args)
{
    int[] arr = { 5, 2, 9, 1, 5, 6 };
    InsertionSort(arr);
    Console.WriteLine(string.Join(", ", arr));
}

File: insertion_sort/insertion_sort.py
def insertionSort2(n, arr):
    # Need to check through the array and print it at each iteration
    print(arr)
    
    for i in range(1, n): # i is the index of the element to be inserted
        key = arr[i] # key is the element to be inserted
        j = i - 1 # j is left of i
        while j >= 0 and arr[j] > key: # while j is in bounds and the element to the left is greater than the key
            arr[j + 1] = arr[j] # shift the element to the right
            j -= 1 # decrement j

        arr[j + 1] = key # insert the key in the correct position
        print (f"Iteration {i}: ", arr) # print the array after each iteration
arr = [12 ,11, 13, 5, 6] # Example array
insertionSort2(5, arr) # Sort the array
print("Sorted array:", arr) # Print the sorted array
